### PR TITLE
Refactor Test Structures 

### PR DESCRIPTION
**Pull request by AlCap23**
Restructure tests and add toolchain / application examples

### CODE REVIEW

**Comment by AlCap23** [comment]
These take a really long time on Travis right now.

Currently this succeeds locally.

**Comment by ChrisRackauckas** [comment]
Split the tests into different environment groups like https://github.com/SciML/DiffEqFlux.jl/blob/master/.travis.yml#L15-L19 and https://github.com/SciML/DiffEqFlux.jl/blob/master/test/runtests.jl#L17

**Comment by AlCap23** [comment]
I think we are ready now. 

There *might* some issues due to the random initialization of the neural network, which can lead to failure.

I think the best way to tackle this is to find either an initial set of parameters and save it or to use bigger networks / hyperopt.
Any worries or ideas on this @ChrisRackauckas ? Else, I'll merge and rebase a whole lot of other PRs in need of this :smile:

File: test/applications/partial_lotka_volterra.jl [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    du[1] = α*u[1] + z[1]
+    du[2] = -δ*u[2] + z[2]
+end
+@info "Simulate system"
+# Create the approximated problem and solution
+ps = [p_[[1,4]]; p̂]
+a_prob = ODEProblem(approx, u0, tspan, ps)
+a_solution = solve(a_prob, Vern7(), abstol=1e-12, reltol=1e-12, saveat = solution.t)
+@test norm(a_solution-solution, 2) < 5e-1
+@test norm(a_solution-solution, Inf) < 1.5e-1
```
Comment (ChrisRackauckas) [inline test/applications/partial_lotka_volterra.jl:163]: can the learned basis be tested to be the quadratic terms?

**Comment by ChrisRackauckas** [comment]
>There might some issues due to the random initialization of the neural network, which can lead to failure.

Yeah, it would probably be good to just isolate that out with some fixed initialization. You can just run it once and then just stick that random vector in the test as the new initial.

**Comment by ChrisRackauckas** [comment]
Because this isn't really about the training, it would probably be good to optimize locally, get the vector, and just stick that in there as the solution and avoid the training part.

File: test/applications/partial_lotka_volterra.jl (hunk 1 above)
Comment (AlCap23) [reply test/applications/partial_lotka_volterra.jl:170]: Yes

**Comment by AlCap23** [comment]
I don't get it right now. There seems to be an issue with the deps, but the lotka volterra test are green for 1.4 with this.

**Comment by AlCap23** [comment]
I excluded the application tests ( which run locally just fine ) for now and updated the compat bounds.

Supersedes #126 ( #112 ), #122 .

Also ups `DiffEqBase` to 6.41 and ``ModelingToolkit` to 3.17

@ChrisRackauckas 
I will merge to get a running master again and try to get the `Adapt` stuff working.

**Pull request merged**